osu.Framework/Platform/HeadlessGameHost.cs

```diff
@@ -30,8 +30,19 @@ public class HeadlessGameHost : DesktopGameHost
 
         public override IEnumerable<string> UserStoragePaths => new[] { "./headless/" };
 
-        public HeadlessGameHost(string gameName = null, bool bindIPC = false, bool realtime = true, bool portableInstallation = false)
-            : base(gameName ?? Guid.NewGuid().ToString(), bindIPC, portableInstallation: portableInstallation)
+        [Obsolete("Use HeadlessGameHost(HostConfig hostConfig) instead.")]
```

Comment:
obsolete notices need updating too

```suggestion
        [Obsolete("Use HeadlessGameHost(HostOptions hostOptions) instead.")]
```

removal dates also missing (goes for more places than this one)

Reply:
I intentionally didn't add/update these until we are ready to agree on this PR (don't think that time has come yet).